model/task_queue.go

```diff
@@ -512,7 +520,30 @@ func FindTaskQueueGenerationTimes() (map[string]time.Time, error) {
 		{
 			"$replaceRoot": bson.M{"newRoot": "$root"},
 		},
-	}, &out)
+	}
+
+}
+
+func FindTaskQueueGenerationTimes() (map[string]time.Time, error) {
+	out := []map[string]time.Time{}
+
+	err := db.Aggregate(TaskQueuesCollection, taskQueueGenerationTimesPipeline(), &out)
+
+	if err != nil {
+		return map[string]time.Time{}, errors.WithStack(err)
```

Comment:
nit: You could return `nil` as the first argument here and in the next return.

Reply:
eh, the callers of this could nil panic, and this is mostly just a guard to avoid over-generating queues, so I'd probably leave it.